### PR TITLE
Add documentaion to markdown

### PR DESCRIPTION
**Pull request by adlerhsieh**

### CODE REVIEW

**Comment by asterite** [comment]
Thanks!

But please mention somewhere that it's very, **very** incomplete. It lacks a lot of features from the official markdown spec.

**Comment by jhass** [comment]
With "spec" you mean Grubers? Still think we should aim for CommonMark eventually ;)

File: src/markdown/markdown.cr [hunk 1]
```diff
@@ -1,3 +1,16 @@
+# Markdown library parses Markdown text. It supports rendering in HTML format.
```
Comment (jhass) [inline src/markdown/markdown.cr]: I think you "render to" something, no "render in" something :)

**Comment by asterite** [comment]
Yes, I did the code reading grubers' spec and his site that generates html from markdown. But CommonMark sounds good too.

File: src/markdown/markdown.cr (hunk 1 above)
Comment (adlerhsieh) [inline src/markdown/markdown.cr]: Yeap that's true, grammatical mistake :P

**Comment by vyp** [comment]
+1 to commonmark.

**Comment by jhass** [comment]
@asterite In case you want to know why CommonMark is important and Gruber didn't really do anything that could be called a "spec" (intentionally so btw): http://johnmacfarlane.net/babelmark2/?text=**foo*+bar* :P

**Comment by adlerhsieh** [comment]
Updated the doc.

File: src/markdown/markdown.cr [hunk 2]
```diff
@@ -1,3 +1,17 @@
+# Markdown library parses Markdown text. It supports rendering to HTML text.
+# Note: This library is in its early stage. Many features and supports are still in development.
```
Comment (jhass) [inline src/markdown/markdown.cr]: Mmh, sorry but using "supports" as a noun in this context sounds weird to me, I associate that to physical structures that support holding something together or making it stronger.

Comment (adlerhsieh) [inline src/markdown/markdown.cr]: Ok, how about just `Many features are still in development.` ?

**Comment by jhass** [comment]
LGTM! Could you squash it into a single commit?

```sh
git remote add upstream https://github.com/manastech/crystal.git
git fetch upstream
git checkout your_branch
git rebase -i upstream/master
# Leave the first commit at pick, choose squash or fix for the others.
# If there are any conflicts git will inform you about that, resolve them for
# example with git mergetool or manually and with git add, then run
# git rebase --continue, again  watch the output since that might happen for
# each commit.
git log # Look at the updated commit history
git status # should report you're back on your branch
# We need to force push since a rebase edits the commits, effectively
# creating new ones. This is okay to do in a feature branch, the pull
# request will update.
git push -f origin your_branch # Assuming you cloned from your fork
```

Hint: You can also edit your last commit with `git commit --amend`, since that also effectively creates a new commit, you to need to force push afterwards and again you should only do it in feature branches.

**Comment by adlerhsieh** [comment]
Thanks for the instruction. It's my first try :+1:  Commits updated.

**Comment by jhass** [comment]
Thanks! For the future create topic branches so you can have multiple PRs open at the same time ;)

**Comment by adlerhsieh** [comment]
Sure, I'll try that out :)

**Pull request merged**